doc/api/esm.md

```diff
@@ -1220,27 +1178,94 @@ loaded from disk but before Node.js executes it; and so on for any `.coffee`,
 `.litcoffee` or `.coffee.md` files referenced via `import` statements of any
 loaded file.
 
-## Resolution algorithm
+#### "import map" loader
+
+The previous two loaders defined `load` hooks. This is an example of a loader
+that does its work via the `resolve` hook. This loader reads an
+`import-map.json` file that specifies which specifiers to override to another
+URL (this is a very simplistic implemenation of a small subset of the
+"import maps" specification).
+
+```js
+// import-map-loader.js
+import fs from 'node:fs/promises';
+
+const { imports } = JSON.parse(await fs.readFile('import-map.json'));
+
+export async function resolve(specifier, context, nextResolve) {
+  if (Object.hasOwn(imports, specifier)) {
+    return nextResolve(imports[specifier], context);
+  }
+
+  return nextResolve(specifier, context);
+}
+```
+
+Let's assume we have these files:
+
+```js
+// main.js
+import 'a-module';
+```
+
+```json
+// import-map.json
+{
+  "imports": {
+    "a-module": "./some-module.js"
+  }
+}
+```
+
+```js
+// some-module.js
+console.log('some module!');
+```
+
+If you run `node --experimental-loader ./import-map-loader.js main.js`
+the output will be `some module!`.
+
+## Default resolution and loading algorithm
 
 ### Features
 
-The resolver has the following properties:
+The default resolver has the following properties:
 
 * FileURL-based resolution as is used by ES modules
-* Support for builtin module loading
 * Relative and absolute URL resolution
 * No default extensions
 * No folder mains
 * Bare specifier package resolution lookup through node\_modules
+* Does not fail on unknown extensions or protocols
+
+The default loader has the following properties
+
+* Support for builtin module loading via `node:` URLs
+* Support for "inline" module loading via `data:` URLs
+* Support for `file:` module loading
+* Fails on any other URL protocol
+* Fails on unknown extensions for `file:` loading
+  (supports only `.cjs`, `.js`, and `.mjs`)
 
-### Resolver algorithm
+### Default resolver algorithm
```

Comment:
Please let's keep this as `Resolver` algorithm. There is _one_ Node.js resolution algorithm.

Reply:
Ah... I can see how the wording is ambiguous.

For me, the resolver algorithm is the algorithm that receives a specifier and parent URL, and calls the chained loaders, where at the end of them sits the _Default resolver_ algorithm. I looked in the `esm.md` above to see what the wording is, and they call it "the Node.js default resolve hook [after the last user-supplied resolve hook[".

So the algorithm outlined here is the algorithm of the "Node.js default resolve hook"., which is what I meant,

Not sure here how to continue and what would be the proper name for this algorithm, because as I said, the Node.js resolver algorithm is the algorithm that calls the chained loaders, and as you said, "Default resolver algorithm" can be interpreted wrongly _and_ may be unclear.

Reply:
You could add a step like "if custom user loaders are registered, pass ___ to the chain of `resolve` hooks; else..."

Reply:
@giltayar technically, you are correct, but that is like saying that a browser is just a JS runtime with a default global environment because you can instrument it adding and removing any custom globals you like. There is still meaning in browser globals that are there by default. A browser global is not a default global environment. The Node.js resolver is not the default Node.js resolver. This is important.